src/binding.cc

```diff
@@ -73,119 +33,117 @@ napi_status napix_get_instance_data(
-  FunctionReference buffer_from;
+  FunctionReference pointer_ctor;
+
+  napi_value WrapPointer(char* ptr, size_t length) override;
+  char* GetBufferData(napi_value val) override;
 
-  void RegisterArrayBuffer(napi_value val) override {
-    ArrayBuffer buf(env, val);
-    RegisterArrayBuffer(buf, AB_PASSED_TO_REF);
+  static InstanceData* Get(Env env) {
+    return env.GetInstanceData<InstanceData>();
   }
+};
 
-  inline void RegisterArrayBuffer(ArrayBuffer buf, ArrayBufferMode mode) {
-    char* ptr = static_cast<char*>(buf.Data());
-    if (ptr == nullptr) return;
+class PointerBuffer : public ObjectWrap<PointerBuffer> {
+ public:
+  static Object Init(Napi::Env env, Object exports);
+  PointerBuffer(Napi::CallbackInfo& info);
+  Napi::Value IsNull(const Napi::CallbackInfo &info);
+  Napi::Value Address(const Napi::CallbackInfo &info);
+  Napi::Value Length(const Napi::CallbackInfo &info);
+  Napi::Value Get(const Napi::CallbackInfo &info);
+  Napi::Value ToString(const Napi::CallbackInfo &info);
+  char *ptr_;
+  int length_;
+};
 
-    auto it = pointer_to_orig_buffer.find(ptr);
-    if (it != pointer_to_orig_buffer.end()) {
-      if (!it->second.ab.Value().IsEmpty()) {
-        // Already have a valid entry, nothing to do.
-        return;
-      }
-      it->second.ab.Reset(buf, 0);
-      it->second.finalizer_count++;
-    } else {
-      pointer_to_orig_buffer.emplace(ptr, ArrayBufferEntry {
-        Reference<ArrayBuffer>::New(buf, 0),
-        1
+Object PointerBuffer::Init(Napi::Env env, Object exports) {
+  Function func =
+      DefineClass(env, "PointerBuffer",{
+           InstanceMethod("isNull", &PointerBuffer::IsNull),
+           InstanceMethod("get", &PointerBuffer::Get),
+           InstanceMethod("address", &PointerBuffer::Address),
+           InstanceMethod("toString", &PointerBuffer::ToString),
+           InstanceAccessor<&PointerBuffer::Length>("length"),
       });
-    }
 
-    // If AB_CREATED_BY_REF, then another finalizer has been added before this
-    // as a "real" backing store finalizer.
-    if (mode != AB_CREATED_BY_REF) {
-      buf.AddFinalizer([this](Env env, char* ptr) {
-        UnregisterArrayBuffer(ptr);
-      }, ptr);
-    }
-  }
+  exports.Set("PointerBuffer", func);
+  InstanceData *data = InstanceData::Get(env);
+  data->pointer_ctor = Persistent(func);
+  return exports;
+}
 
-  inline void UnregisterArrayBuffer(char* ptr) {
-    auto it = pointer_to_orig_buffer.find(ptr);
-    if (--it->second.finalizer_count == 0)
-      pointer_to_orig_buffer.erase(it);
+PointerBuffer::PointerBuffer(Napi::CallbackInfo& info) : Napi::ObjectWrap<PointerBuffer>(info) {
+  Napi::Env env = info.Env();
+  int length = info.Length();
+  if (length <= 0 || !info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Number expected").ThrowAsJavaScriptException();
+    return;
   }
+  Napi::Number value = info[0].As<Napi::Number>();
+  ptr_ = reinterpret_cast<char *>(value.Int64Value());
```

Comment:
The JS number type can’t accurately represent 64-bit values, so this will not work as-is, I am afraid :/